addons/crm/tests/test_crm_lead.py

```diff
@@ -386,3 +386,24 @@ def test_mailgateway(self):
         new_lead._handle_partner_assignment(create_missing=True)
         self.assertEqual(new_lead.partner_id.email, 'unknown.sender@test.example.com')
         self.assertEqual(new_lead.partner_id.team_id, self.sales_team_1)
+
+    def test_multicompany_team(self):
```

Comment:
Would probably make a new class TestCRMLeadMC with a small setup with another company, and add this test in this class, so that we may add some MC tests in an hypothetical future.